tools/@aws-cdk/spec2cdk/lib/cdk/arn.ts

```diff
@@ -7,19 +7,22 @@ import { Resource } from '@aws-cdk/service-spec-types';
  * included in the primary identifier.
  */
 export function findNonIdentifierArnProperty(resource: Resource) {
-  return findArnProperty(resource, (name) => !resource.primaryIdentifier?.includes(name));
+  const refIdentifier = resource.cfnRefIdentifier ?? resource.primaryIdentifier;
+  return findArnProperty(resource, (name) => !refIdentifier?.includes(name));
```

Comment:
i know you didn't add in the `!`, but if we're confident `primaryIdentifier` always exists, then why is the property typed as optional in the first place?

Reply:
I think we're confident that it exists *here*, not that it exists in general?

The property is probably optional because we added it later.